browser/tor/tor_profile_service_impl.cc

```diff
@@ -6,28 +6,69 @@
 #include "brave/browser/tor/tor_profile_service_impl.h"
 
 #include <string>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/task/post_task.h"
+#include "brave/browser/brave_browser_process_impl.h"
+// TODO(bridiver) - move this out of extensions
+#include "brave/browser/extensions/brave_tor_client_updater.h"
 #include "brave/browser/tor/tor_launcher_service_observer.h"
+#include "brave/common/tor/pref_names.h"
+#include "brave/common/tor/tor_constants.h"
+#include "brave/net/proxy_resolution/proxy_config_service_tor.h"
 #include "chrome/browser/profiles/profile.h"
+#include "components/prefs/pref_service.h"
 #include "content/public/browser/browser_task_traits.h"
 #include "content/public/browser/browser_thread.h"
 #include "content/public/browser/site_instance.h"
 #include "content/public/browser/storage_partition.h"
+#include "mojo/public/cpp/bindings/interface_request.h"
 #include "net/proxy_resolution/proxy_resolution_service.h"
 #include "net/url_request/url_request_context.h"
-#include "net/url_request/url_request_context_getter.h"
+#include "services/network/public/mojom/network_context.mojom.h"
 
 using content::BrowserContext;
 using content::BrowserThread;
 
 namespace tor {
 
-TorProfileServiceImpl::TorProfileServiceImpl(Profile* profile) :
-    profile_(profile) {
+namespace {
+
+void OnProfileCreatedOnIOThread(Profile* profile, scoped_refptr<net::URLRequestContextGetter> request_context) {
```

Comment:
we should avoid using `URLRequestContextGetter`, since it doesn't exist when `NetworkService` is enabled

Reply:
oh, that complicates things... hopefully there is another way to get the ProxyResolutionService

Reply:
or maybe we can use NetworkTrafficAnnotationTag?

Reply:
actually I think I know how to get around it